src/QueryType/Server/Collections/Result/ZkStateReader.php

```diff
@@ -0,0 +1,774 @@
+<?php
+/**
+ * BSD 2-Clause License
+ *
+ * Copyright (c) 2017 Jeroen Steggink
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ *  Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Solarium\QueryType\Server\Collections\Result;
+
+use DateInterval;
+use Solarium\Cloud\Core\Client\CollectionEndpoint;
+use Solarium\Cloud\Exception\SolrCloudException;
+use Solarium\Cloud\Exception\ZookeeperException;
+use Solarium\Exception\InvalidArgumentException;
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Zookeeper;
```

Comment:
Oops, this accidently ended up in a commit. This is code from the other project https://github.com/solariumphp/solarium-cloud

Reply:
From my point of view, Solr Cloud becomes the standard. For example all these streaming expressions require Solr Cloud.
So we should either move streaming expressions to solarium-cloud or integrate solarium-cloud here.
I vote for the second option ;-)
Again, I already worked in the same direction and would love replace it by your code.

Reply:
@jsteggink may I invite you to https://drupalchat.me/channel/solarium-php

It would be nice to be able to chat about some stuff. The European Drupal community is so kind to host us :-)